Firestore/core/src/firebase/firestore/auth/firebase_credentials_provider_apple.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Right now, FirebaseCredentialsProvider only support APPLE build.
+#if !defined(__OBJC__)
+#error "This .header can only be used in .mm file for iOS build."
```

Comment:
".header" reads strangely also, this isn't just iOS, but also macOS, tvOS, etc

Perhaps:

```c
#errror "This header only supports Objective-C++"
```